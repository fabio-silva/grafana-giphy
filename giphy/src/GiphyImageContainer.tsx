import React from 'react';
import { stylesFactory } from '@grafana/ui';
import { css, cx } from 'emotion';
import { GiphyImage } from 'GiphyImage';
import { ImageData } from 'types';

const styles = stylesFactory(() => ({
  giphyImg: css`
    height: 150px;
    margin-bottom: 10px;
    &:not(:last-child) {
      margin-right: 10px;
    }
	`
}))();

interface ImageProps {
  images: ImageData[] | null
};

export const GiphyImageContainer: React.FC<ImageProps> = ({ images }) => {
  let imageEl: JSX.Element[] | JSX.Element = <></>;
  // At this point, we can check if we should show a "no results found"
  if (images) {
    if (images.length) {
      imageEl = images.map(({ id, url }) => <div
        className={cx(styles.giphyImg)}
        key={id}
      >
        <GiphyImage url={url} />
      </div>
      )
    } else {
      imageEl = <p>No result found</p>;
    }
  }

  return <>{imageEl}</>;
}