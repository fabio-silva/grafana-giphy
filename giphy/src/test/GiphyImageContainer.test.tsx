import React from 'react';
import { shallow } from 'enzyme';
import { GiphyImageContainer } from '../GiphyImageContainer';

describe('GiphyImageContainer', () => {
  it('should display "No result found when getting empty array"', () => {
    const container = shallow(<GiphyImageContainer images={[]}/>);
    expect(container.html().includes('No result found')).toBe(true);
  });

  it('should display an empty node when null is passed', () => {
    const container = shallow(<GiphyImageContainer images={null}/>);
    expect(container.html().length).toBe(0);
  });

  it('should render one GiphyImage per image passed', () => {
    const images = [{
      id: '1',
      url: 'url_1'
    }, {
      id: '2',
      url: 'url_2'
    }];
    const container = shallow(<GiphyImageContainer images={images}/>);
    expect(container.children().length).toBe(2);
    expect(container).toMatchSnapshot();
  });
});
