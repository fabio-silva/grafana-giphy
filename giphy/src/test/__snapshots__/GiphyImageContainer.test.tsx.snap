// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiphyImageContainer should one GiphyImage per image passed 1`] = `
<Fragment>
  <div
    className="css-1mzojan"
    key="1"
  >
    <Component
      url="url_1"
    />
  </div>
  <div
    className="css-1mzojan"
    key="2"
  >
    <Component
      url="url_2"
    />
  </div>
</Fragment>
`;
